client/txn.go

```diff
@@ -76,6 +76,7 @@ type Txn struct {
 	// is (still) unset.
 	haveTxnWrite bool
 	haveEndTxn   bool // True if there was an explicit EndTransaction
+	execing      bool // True if the txn is run in the exec() loop
```

Comment:
You always want `Commit` externally, except for that test you mentioned. At least I think that's true, based on your previous comment.